src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -85,7 +95,90 @@ public async Task RunApplicationAsync(CancellationToken cancellationToken = defa
         {
             AspireEventSource.Instance.DcpModelCreationStop();
         }
+    }
+
+    private bool TryGetMetadataPath(string key, [NotNullWhen(true)]out string? path)
+    {
+        var entryAssembly = Assembly.GetEntryAssembly()!; // Null suppression because it can't return null in this context.
+        var metadataAttributes = entryAssembly.GetCustomAttributes<AssemblyMetadataAttribute>();
+
+        if (metadataAttributes.SingleOrDefault(a => a.Key == key) is not { } pathAttribute)
+        {
+            _logger.LogWarning("Metadata attribute with key '{Key}' is not found in assembly '{EntryAssembly}'", key, entryAssembly.FullName);
+            path = null;
+            return false;
+        }
+
+        if (string.IsNullOrEmpty(pathAttribute.Value))
+        {
+            _logger.LogWarning("Metadata attribute with key '{Key}' is empty in assembly '{EntryAssembly}'", key, entryAssembly.FullName);
+            path = null;
+            return false;
+        }
+
+        var fullyQualifiedPath = Path.GetFullPath(pathAttribute.Value);
+
+        if (!File.Exists(fullyQualifiedPath))
+        {
+            path = null;
+            return false;
+        }
+
+        path = fullyQualifiedPath;
+        return true;
+    }
+
+    private const string DashboardPathMetadataKey = "aspiredashboardpath";
+
+    private async Task StartDashboardAsync(CancellationToken cancellationToken = default)
+    {
+        if (!TryGetMetadataPath(DashboardPathMetadataKey, out var dashboardPath))
+        {
+            _logger.LogWarning("Dashboard is not being started because path was not present or invalid.");
+            return;
+        }
+
+        var dashboardWorkingDirectory = Path.GetDirectoryName(dashboardPath);
+
+        var dashboardExecutableSpec = new ExecutableSpec();
```

Comment:
Why did you decide to build the ExecutableSpec directly vs creating an executable resource, then converting it to a spec?

I could imagine it is useful to have some logic like logging and OTEL env vars automatically set - which is done at the resource layer - without lots of duplication.

Reply:
Because we don't want the dashboard showing up as a resource. There is presently a bug in the dashboard where it is showing the aspire-dashboard executable because it is querying the DCP resource list instead of using the app model resources.

Reply:
> showing up as a resource

Is this a UI thing? What about adding support for an internal flag that says a resource shouldn't be displayed in the dashboard UI.

Reply:
Maybe what this PR is doing now is the right thing, but I'd like to understand why it needs special treatment.

Reply:
We just need to filter it out on the resource server side of things.

Reply:
The logic is that the dashboard is not a resource as far as the app model is concerned. It is more like `dcp.exe`. It doesn't get referenced by other resources, in fact it needs to be spun up before other resources to avoid missing telemetry (side note, this PR does not yet contain a check to make sure that the dashboard has started).

Reply:
The resource server currently queries DCP so this doesn't do anything AFAIK. Doesn't it still show up? If you still need to filter it out, what does this buy you?

Reply:
It still shows up because its a bug. Dashboard shouldn't be querying DCP directly, it should be filtered through the lens of the app model. Why is it important for the dashboard to be an Aspire resource?

Reply:
If we want to go down the path of making this a resource we might want to add another lifeycle callback method (something like `AfterStartAsync(...)`) where we can inject resourecs after the `DashboardServiceHost` is started but before we start allocating endpoints. Then once we've got the dashboard in the resource list - when we go to start them in DCP we would need to order them (or pick out the ones with aspire-dashboard naming specifically) and start them before the others so that the OTLP endpoint is available.

I already have to add the logic to check that the dashboard is alive before allowing execution to continue.

Reply:
> It still shows up because its a bug. Dashboard shouldn't be querying DCP directly, it should be filtered through the lens of the app model. Why is it important for the dashboard to be an Aspire resource?

I'm not sure if it is, but that'll make it easy to debug etc 😄. I'm OK making it a DCP resource directly, its not like the app needs to interact with it. That said, we are not changing the resource server logic to do the right thing and use the app model yet. Not until we get this end to end working as well as preview2 with all of the refactoring.

Reply:
This PR now includes logic to filter out aspire-dashboard* resources. Still some tidy up to do on that but its working.